libraries/AP_GPS/GPS_Backend.cpp

```diff
@@ -112,3 +114,38 @@ void AP_GPS_Backend::fill_3d_velocity(void)
     state.velocity.z = 0;
     state.have_vertical_velocity = false;
 }
+
+
+void AP_GPS_Backend::_detection_message(char *buffer, const uint8_t buflen) const
+{
+    const uint8_t instance = state.instance;
+    const struct AP_GPS::detect_state dstate = gps.detect_state[instance];
+
+    if (dstate.detect_started_ms > 0) {
+        hal.util->snprintf(buffer, buflen,
+                 "GPS %d: detected as %s at %d baud",
+                 instance + 1,
+                 name(),
+                 gps._baudrates[dstate.current_baud]);
+    } else {
+        hal.util->snprintf(buffer, buflen,
+                 "GPS %d: specified as %s",
```

Comment:
With the first commits, it made sense to have *specified* as the alternative message, but since changing PX4 driver to only send the message when actually detected, this message doesn't apply so well for it.

I'm not sure how we can do better without over complicating though - also with the CAN work that is being done, the PX4 driver will be removed soon. So maybe this is fine?